tests/test_enki.py

```diff
@@ -364,31 +410,76 @@ def test_checkout_not_available(self):
 
         loan = self.api.checkout(patron,'1234',pool,None)
 
-    def test_fulfillment_open_access(self):
+    def test_fulfillment_open_access_parser(self):
         """Test that fulfillment info for non-ACS Enki books is parsed correctly."""
         data = self.get_data("checked_out_direct.json")
-        self.api.queue_response(200, content=data)
         result = json.loads(data)
         fulfill_data = self.api.parse_fulfill_result(result['result'])
         eq_(fulfill_data[0], """http://cccl.enkilibrary.org/API/UserAPI?method=downloadEContentFile&username=21901000008080&password=deng&lib=1&recordId=2""")
         eq_(fulfill_data[1], 'epub')
 
-    def test_fulfillment_acs(self):
+    def test_fulfillment_acs_parser(self):
         """Test that fulfillment info for ACS Enki books is parsed correctly."""
         data = self.get_data("checked_out_acs.json")
-        self.api.queue_response(200, content=data)
         result = json.loads(data)
         fulfill_data = self.api.parse_fulfill_result(result['result'])
         eq_(fulfill_data[0], """http://afs.enkilibrary.org/fulfillment/URLLink.acsm?action=enterloan&ordersource=Califa&orderid=ACS4-9243146841581187248119581&resid=urn%3Auuid%3Ad5f54da9-8177-43de-a53d-ef521bc113b4&gbauthdate=Wed%2C+23+Aug+2017+19%3A42%3A35+%2B0000&dateval=1503517355&rights=%24lat%231505331755%24&gblver=4&auth=8604f0fc3f014365ea8d3c4198c721ed7ed2c16d""")
         eq_(fulfill_data[1], 'epub')
 
+    def test_fulfill_success(self):
+        # Test the fulfill() method.
+        patron = self._patron()
+        patron.authorization_identifier = "123"
+        pool = self._licensepool(None)
+
+        data = self.get_data("checked_out_acs.json")
+        self.api.queue_response(200, content=data)
+        fulfillment = self.api.fulfill(patron, "pin", pool, "internal format")
+
+        # An appropriate request to the "getSELink" endpoint was made.,
+        [method, url, headers, data, params, kwargs] = self.api.requests.pop()
+        eq_("get", method)
+        eq_(self.api.base_url + "UserAPI", url)
+        eq_("getSELink", params['method'])
+        eq_("123", params['username'])
+        eq_("pin", params['password'])
+
+        # In particular, the Enki library ID associated with the
+        # patron's library was used as the 'lib' parameter.
+        eq_("c", params['lib'])
+
+        # A FulfillmentInfo for the loan was returned.
+        assert isinstance(fulfillment, FulfillmentInfo)
+        eq_(fulfillment.identifier, pool.identifier.identifier)
+        eq_(fulfillment.collection_id, pool.collection.id)
+        eq_(DeliveryMechanism.ADOBE_DRM, fulfillment.content_type)
+        assert fulfillment.content_link.startswith(
+            "http://afs.enkilibrary.org/fulfillment/URLLink.acsm"
+        )
+        eq_(fulfillment.content_expires,
+            datetime.datetime(2017, 9, 13, 19, 42, 35, 0))
+
     def test_patron_activity(self):
```

Comment:
This test verified the effects of calling `patron_activity` but not the request that went over the network.